### PR TITLE
📖  Add doc for e2e pipeline

### LINKED ISSUE #91: Add documentation for e2e pipeline
/kind feature

We would like to add documentation for the e2e pipeline, similar to [this one](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/blob/main/doc/continuous-integration.md).

### PR DESCRIPTION
**Pull request by michal-gubricky**
**What this PR does / why we need it**:
This PR adds documentation for the e2e pipeline, check [PR#53](https://github.com/SovereignCloudStack/cluster-stacks/pull/53)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #91

### CODE REVIEW

File: doc/continuous-integration.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    that ensures that only one `openstack-access-base` job (or its children) can run at a time
+  - See a high level `e2e-openstack-conformance` job steps:
+    - Pre-run playbook `dependencies.yaml` installs project prerequisites, e.g. clusterctl, KinD, csctl, etc.
+    - Main playbook `e2e.yaml` spawns a k8s workload cluster using a specific cluster-stack, runs sonobuoy conformance test, SCS compliance test, and cleans created k8s workload cluster
+
+- `e2e-openstack-quick`
+  - It runs a sonobuoy quick test against Kubernetes cluster spawned by a specific cluster-stack
+  - This job is a child job of `openstack-access-base` that ensures OpenStack credentials
+    availability in Zuul worker node. Parent job also defines a Zuul semaphore `semaphore-openstack-access`,
+    that ensures that only one `openstack-access-base` job (or its children) can run at a time
```
[the remark this answers is not in the document]
Comment (michal-gubricky) [reply doc/continuous-integration.md:89]: Then also doc in `k8s-cluster-api-provider` repo should be updated.

[the remark this answers is not in the document]
Comment (chess-knight) [reply doc/continuous-integration.md]: Would you like to do it?

[the remark this answers is not in the document]
Comment (michal-gubricky) [reply doc/continuous-integration.md]: I have no preference, but I can do it.

[the remark this answers is not in the document]
Comment (chess-knight) [reply doc/continuous-integration.md]: Ok

**Review by chess-knight — APPROVED** [review]

**Pull request merged**